src/core_plugins/interpreter/public/interpreter.js

```diff
@@ -0,0 +1,53 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { socketInterpreterProvider } from '@kbn/interpreter-utils/common/interpreter/socket_interpret';
+import { serializeProvider } from '@kbn/interpreter-utils/common/lib/serialize';
+import { typesRegistry } from '@kbn/interpreter-utils/common/lib/types_registry';
+import { createHandlers } from '@kbn/interpreter-utils/ui/lib/create_handlers';
+import { functionsRegistry } from '@kbn/interpreter-utils/ui/lib/functions_registry';
+import { socket } from './socket';
+
+// Create the function list
+socket.emit('getFunctionList');
+export const getServerFunctions = new Promise(resolve => socket.once('functionList', resolve));
+
+// Use the above promise to seed the interpreter with the functions it can defer to
+export function interpretAst(ast, context) {
+  // Load plugins before attempting to get functions, otherwise this gets racey
```

Comment:
we no longer wait for loadBrowserPlugins as we expect them to be loaded at this time